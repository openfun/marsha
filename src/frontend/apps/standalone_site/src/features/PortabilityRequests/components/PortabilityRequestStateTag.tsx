import { Tag } from 'grommet';
import { normalizeColor } from 'grommet/utils';
import { theme } from 'lib-common';
import { PortabilityRequestState } from 'lib-components';
import { defineMessages, useIntl } from 'react-intl';
import styled from 'styled-components';

const NeutralStyledTag = styled(Tag)`
  color: ${normalizeColor('blue-active', theme)};
  border-color: ${normalizeColor('blue-active', theme)};
`;

const StatusOkStyledTag = styled(Tag)`
  color: ${normalizeColor('status-ok', theme)};
  border-color: ${normalizeColor('status-ok', theme)};
`;

const StatusErrorStyledTag = styled(Tag)`
  color: ${normalizeColor('status-error', theme)};
  border-color: ${normalizeColor('status-error', theme)};
`;

const messages = defineMessages({
  pendingState: {
    defaultMessage: 'pending',
    description: 'Pending display state',
    id: 'features.PortabilityRequestStateTag.pendingState',
  },
  acceptedState: {
    defaultMessage: 'accepted',
    description: 'Accepted display state',
    id: 'features.PortabilityRequestStateTag.acceptedState',
  },
  rejectedState: {
    defaultMessage: 'rejected',
    description: 'Rejected display state',
    id: 'features.PortabilityRequestStateTag.rejectedState',
  },
});

export const PortabilityRequestStateTag = ({
  state,
}: {
  state: PortabilityRequestState;
}) => {
  const intl = useIntl();

  switch (state) {
    case PortabilityRequestState.PENDING:
      return (
        <NeutralStyledTag
          alignSelf="center"
          value={intl.formatMessage(messages.pendingState)}
        />
      );
    case PortabilityRequestState.ACCEPTED:
      return (
        <StatusOkStyledTag
          alignSelf="center"
          value={intl.formatMessage(messages.acceptedState)}
        />
      );
    case PortabilityRequestState.REJECTED:
      return (
        <StatusErrorStyledTag
          alignSelf="center"
          value={intl.formatMessage(messages.rejectedState)}
        />
      );
    default:
      return <NeutralStyledTag alignSelf="center" value={state} />;
  }
};
