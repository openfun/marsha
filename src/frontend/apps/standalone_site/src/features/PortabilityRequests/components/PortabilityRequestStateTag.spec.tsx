import { screen } from '@testing-library/react';
import { PortabilityRequestState } from 'lib-components';
import { render } from 'lib-tests';

import { PortabilityRequestStateTag } from './PortabilityRequestStateTag';

describe('<PortabilityRequestStateTag />', () => {
  it('renders pending state', () => {
    render(
      <PortabilityRequestStateTag state={PortabilityRequestState.PENDING} />,
    );
    expect(screen.getByText('pending')).toBeInTheDocument();
  });

  it('renders accepted state', () => {
    render(
      <PortabilityRequestStateTag state={PortabilityRequestState.ACCEPTED} />,
    );
    expect(screen.getByText('accepted')).toBeInTheDocument();
  });

  it('renders rejected state', () => {
    render(
      <PortabilityRequestStateTag state={PortabilityRequestState.REJECTED} />,
    );
    expect(screen.getByText('rejected')).toBeInTheDocument();
  });
});
