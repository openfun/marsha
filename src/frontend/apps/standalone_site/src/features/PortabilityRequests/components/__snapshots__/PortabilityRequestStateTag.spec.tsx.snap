// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortabilityRequestStateTag /> accepted state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-hhOBVt PoHIa"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              accepted
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> pending state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-jIRcFI enNWMs"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              pending
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> rejected state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-gYbzsP liRoWs"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              rejected
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> renders accepted state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-hhOBVt PoHIa"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              accepted
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> renders pending state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-jIRcFI enNWMs"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              pending
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> renders rejected state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-gYbzsP liRoWs"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              rejected
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PortabilityRequestStateTag /> rendersaccepted state 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jaywUP"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      data-testid="test-component-container"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bNHvXa sc-hhOBVt PoHIa"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eYzGpf"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jbJJSg"
          >
            
            <span
              class="StyledText-sc-1sadyjn-0 iGZjmp"
            >
              accepted
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
